src/devices/src/virtio/vsock/device.rs

```diff
@@ -47,22 +49,63 @@ pub(crate) const EVQ_INDEX: usize = 2;
 pub(crate) const AVAIL_FEATURES: u64 =
     1 << uapi::VIRTIO_F_VERSION_1 as u64 | 1 << uapi::VIRTIO_F_IN_ORDER as u64;
 
-pub struct Vsock<B> {
-    cid: u64,
-    pub(crate) queues: Vec<VirtQueue>,
+pub struct Vsock<B: 'static> {
     pub(crate) queue_events: Vec<EventFd>,
     pub(crate) backend: B,
-    avail_features: u64,
-    acked_features: u64,
-    pub(crate) interrupt_status: Arc<AtomicUsize>,
     pub(crate) interrupt_evt: EventFd,
     // This EventFd is the only one initially registered for a vsock device, and is used to convert
     // a VirtioDevice::activate call into an EventHandler read event which allows the other events
     // (queue and backend related) to be registered post virtio device activation. That's
     // mostly something we wanted to happen for the backend events, to prevent (potentially)
     // continuous triggers from happening before the device gets activated.
     pub(crate) activate_evt: EventFd,
-    pub(crate) device_state: DeviceState,
+    pub(crate) device_status: DeviceStatus,
+    pub(crate) state: VsockState,
+}
+
+/// A helper structure that holds the usual constructor parameters that are not serialized in VsockState
+pub struct VsockConstructorArgs<B> {
+    mem: GuestMemoryMmap,
+    backend: B,
+}
+
+/// The Vsock serializable state.
+#[derive(Clone, Versionize)]
+pub struct VsockState {
+    cid: u64,
+    queues: Vec<VirtQueue>,
```

Comment:
I think it is better for VirtQueue to have an associated state. This way, you don't need to implement Versonize for GuestAddress. In the state, you can simply save the underlying `u64`.

Reply:
GuestAddress is basically an u64. The benefit of having Versionize implemented for GuestAddress is that you do not need to create a separate state struct for VirtuQueue or any other POD state that has GuestAddress members.  Please see my above reply about defining a clear way of separating POD states from live state. There is a lot of benefit in readability and consistency when relying on code generation where this doesn't hinder the separation of concerns.